src/System.Net.Security/src/System/Net/Security/SslStreamInternal.cs

```diff
@@ -231,23 +209,13 @@ private void EnsureInternalBufferSize(int newSize)
                 bool wasPinnable = _internalBufferFromPinnableCache;
                 byte[] saved = _internalBuffer;
 
-                if (newSize <= PinnableReadBufferSize)
+                if (newSize <= ReadBufferSize)
```

Comment:
In what circumstance is newSize > ReadBufferSize?  And if it's common, why can't we use the pool for that, too? i.e. why is there a limit?

Reply:
None, and actually we end up with a 32k buffer often which we could better take advantage of (2^n for array pool buffers and as we are 2^14 and change we go to 2^15).  I honestly didn't massively look at the read side until last night as I have focused on the write, but there are some quick wins so I will fix these later today.